src/uast.h

```diff
@@ -1,20 +1,70 @@
 #ifndef LIBUAST_UAST_H_
 #define LIBUAST_UAST_H_
 
+#include "export.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stddef.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "export.h"
-#include "node_iface.h"
-#include "nodes.h"
-
 // Uast stores the general context required for library functions.
 // It must be initialized with `UastNew` passing a valid implementation of the
 // `NodeIface` interface.
 // Once it is not used anymore, it shall be released calling `UastFree`.
 typedef struct Uast Uast;
 
+typedef uintptr_t NodeHandle;
```

Comment:
Why was this moved from `node_iface.h`?

Reply:
Because `node_iface.h` requires `Uast*` that is defined in `uast.h`, and `uast.h` requires `NodeIface`.